tests/unit/test_reddit.py

```diff
@@ -76,13 +76,13 @@ def test_context_manager(self):
 
     def test_info__invalid_param(self):
         with pytest.raises(TypeError) as excinfo:
-            self.reddit.info(None)
+            next(self.reddit.info(None))
```

Comment:
Oh so the behavior is different. This will end up being a breaking change. Not that it's bad, but we'd have to announce the change first. Maybe that was the reason for returning the generator... so that validations would run immediately rather than deferring until iterating upon.